File: OOPS/oopsJava/abstraction.encapsulation/AbstractionExample.java
package in.kpriet.training.oops;

abstract class Shape {

    String color;

    Shape(String color) {
        this.color = color;
    }

    public abstract double getArea(double number1, double number2);
    
    public String getColor() {
        return this.color;
    }
}


class Triangle extends Shape {

    Triangle(String color) {
        super(color);
    }

    public double getArea(double number1, double number2) {
        return 0.5 * number1 * number2;
    }
}


class Rectangle extends Shape {

    Rectangle(String color) {
        super(color);
    }

    public double getArea(double number1, double number2) {
        return number1 * number2;
    }
}


class AbstractionExample {

    public static void main(String[] args) {
        
        Shape triangle = new Triangle("Black");
        Shape rectangle = new Rectangle("Blue");

        System.out.println(triangle.getArea(3, 5));
        System.out.println(rectangle.getArea(4, 5));

        System.out.println(triangle.getColor());
        System.out.println(rectangle.getColor());
    }
}

File: OOPS/oopsPython/abstraction.encapsulation/AbstractionExample.py
from abc import ABC, abstractmethod

class Shape(ABC):

    def __init__(self, color):
        self.color = color
        
    @abstractmethod
    def getArea(self, number1, number2):
        pass

    def getColor(self):
        return self.color
    

class Triangle(Shape):

    def getArea(self, number1, number2):
        return 0.5 * number1 * number2


class Rectangle(Shape):

    def getArea(self, number1, number2):
        return number1 * number2


triangle = Triangle("black")
rectangle = Rectangle("blue")

print(triangle.getArea(2, 5))
print(rectangle.getArea(5, 7))

print(triangle.getColor())
print(rectangle.getColor())
